### PR TITLE
Fix datetime formatting for pulse/dashboard subscription attachments

### PR DESCRIPTION
**Pull request by noahmoss**
When queries are run for pulses and dashboard subscriptions, they include the `format-rows` middleware which formats temporal values as strings. This causes formatting for these rows to be excluded from XLSX attachments.

We can't omit the `format-rows` middleware since downstream static viz code depends on it, for queries that don't result in attachments.

My solution is to parse these strings back into timestamps, _if_ the semantic type of the column is temporal, before writing them to the XLSX document.

### CODE REVIEW

File: src/metabase/pulse.clj [hunk 1]
```diff
@@ -76,7 +77,6 @@
                     :dashboard-id  (:id dashboard)
                     :context       :pulse ; TODO - we should support for `:dashboard-subscription` and use that to differentiate the two
                     :export-format :api
-                    :middleware    {:process-viz-settings? true}
```
Comment (noahmoss) [inline src/metabase/pulse.clj:79]: Removed this line since it's redundant

**Review by mazameli — COMMENTED** [review]
Worked for `xlsx` but not `csv` which I think is expected. 

Oddly didn't work for my first column, `Birth Date`, at least not as expected. The date style I chose for it was `7 January, 2018`

image


image

This does however excentuate the fact that the inline table in the email itself does not respect the formatting:

image

**Comment by noahmoss** [comment]
Okay, I've changed the approach a bit: now I'm attempting to parse _all_ strings as datetime values during export and falling back to the plain strings if that fails. This should fix formatting for columns that contain temporal data, but don't have a temporal semantic type assigned.

I've restricted this extra parsing logic to exports that are triggered by pulses/dashboard subscriptions, not normal user-requested exports, so there shouldn't be extra latency for the latter.

@mazameli if you get a chance, mind testing your example again and seeing if it now works for the `Birth Date` column?

**Review by camsaul — APPROVED** [review]
backend stuff LGTM

**Pull request merged**